include/dali/core/expand_dims.h

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_EXPAND_DIMS_H_
+#define DALI_CORE_EXPAND_DIMS_H_
+
+#include "dali/core/tensor_shape.h"
+
+namespace dali {
+
+/**
+ * @brief  Inserts new dimensions with extent one to the new shape as needed to match a given number
```

Comment:
Nitpick:
```suggestion
 * @brief  Inserts new dimensions with extent 1 to the new shape as needed to match a given number
```
I think it looks a tiny bit less WTF this way.